### ISSUE TITLE
No oauth window appear after clicking the Google sign-in button

### ISSUE THREAD

**Issue by martgil** [FlowCrypt/flowcrypt-browser#4553]
Description:
There is no OAuth window appearing when the user clicks on the Google sign-in button. I'm unable to find out the root cause of it for the moment.

From the user:
>  I just reinstalled Chrome, and I'm trying to set it up
again.
When I click the "Sign in with Google" button and expect to see the
oAuth2 popup window, I get the attached error message pop up instantly
instead. There is no oAuth window. It doesn't matter what Google
account I try to select in the upper right corner either. 

screenshot:
![image](https://user-images.githubusercontent.com/46025304/179951436-263a9dcc-9f51-4f7b-841c-7d64fa5e226d.png)

<details><summary>stack traces: </summary>

```
DevTools failed to load source map: Could not load content for chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/forge.js.map: System error: net::ERR_FILE_NOT_FOUND
DevTools failed to load source map: Could not load content for chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/bootstrap/bootstrap.min.js.map: System error: net::ERR_FILE_NOT_FOUND
DevTools failed to load source map: Could not load content for chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/bootstrap/bootstrap.min.css.map: System error: net::ERR_FILE_NOT_FOUND
catch.js:60 Error: failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
    at Catch.nameAndDetailsAsException (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:345:15)
    at Catch.report (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:98:34)
    at Settings.newGoogleAcctAuthPromptThenAlertOrForward (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/settings.js:361:23)
    at async SettingsView.<anonymous> (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/index.js:201:28)


details:
[typeof:object:[object Object]] {"result":"Error","error":"No oauth window returned after initiating it","acctEmail":"some_email@domain.test"}
Catch.onErrorInternalHandler @ catch.js:60
Catch.reportErr @ catch.js:92
Catch.report @ catch.js:98
Settings.newGoogleAcctAuthPromptThenAlertOrForward @ settings.js:361
await in Settings.newGoogleAcctAuthPromptThenAlertOrForward (async)
(anonymous) @ index.js:201
cbWithErrsHandled @ ui.js:74
(anonymous) @ ui.js:110
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
catch.js:64 failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
Error: failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
    at Catch.nameAndDetailsAsException (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:345:15)
    at Catch.report (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:98:34)
    at Settings.newGoogleAcctAuthPromptThenAlertOrForward (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/settings.js:361:23)
    at async SettingsView.<anonymous> (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/index.js:201:28)


details:
[typeof:object:[object Object]] {"result":"Error","error":"No oauth window returned after initiating it","acctEmail":"some_email@domain.test"}
Catch.onErrorInternalHandler @ catch.js:64
Catch.reportErr @ catch.js:92
Catch.report @ catch.js:98
Settings.newGoogleAcctAuthPromptThenAlertOrForward @ settings.js:361
await in Settings.newGoogleAcctAuthPromptThenAlertOrForward (async)
(anonymous) @ index.js:201
cbWithErrsHandled @ ui.js:74
(anonymous) @ ui.js:110
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
index.htm:1 Unchecked runtime.lastError: Invalid value for bounds. Bounds must be at least 50% within visible screen space.
catch.js:291 FlowCrypt ERROR: Please report errors above to human@flowcrypt.com. We fix errors VERY promptly.
```

</details>

Reference: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGpGwmbmBXStgffvgCpbjFPQQCV

**Comment by martgil** [comment]
I tried disabling javascript and disallowing pop-ups, thinking that I can replicate the issue, but, none of them work so far.

**Comment by neilschelly** [comment]
I'm the user in question. I will provide addition details if it helps.

1. I am using Chrome Version 103.0.5060.114 (Official Build) (64-bit) on Debian Linux Bullseye (fully updated).
2. I wiped out my google chrome profile directory to recreate it from scratch. FlowCrypt was setup before I did this.
3. When I logged into my Chrome profile again, all my extensions got reinstalled, including FlowCrypt. This problem was apparent immediately.
4. I have removed the authorization for FlowCrypt from my Google Account settings.
5. I have tried disabling all extensions except FlowCrypt.
6. I have tried removing and reinstalling the FlowCrypt extension while all other extensions were disabled.
7. When I click the Setup button after installation, I get the page in the screenshot above (with no error yet), except there are not any google accounts listed in the upper right yet.
8. When I click the "Set up FlowCrypt now for encrypted email?" prompt in GMail, I get to the screenshot above (with no error yet), with my google accounts in the menu to the upper right.
9. When I click the "Sign in with Google" button above, the error pops up immediately and there's no network activity until it retrieves the error message itself.
10. It doesn't matter which Google Account I select from the menu at the upper right. Historically, only one of them ever had FlowCrypt setup anyway.

Hope that helps...

**Comment by neilschelly** [comment]
More context...
1. I completely uninstalled Google Chrome, removed my profile and cache directories, etc.
2. I reinstalled Chrome from the Google site (currently 103.0.5060.134-1).
3. I started it up, didn't login to Chrome (starting with empty profile), navigated to the FlowCrypt website to install the extension.
4. I tried the Sign in with Google button, and got the same error as above.
5. I retried the above steps with google-chrome-beta (currently 104.0.5112.57-1) and google-chrome-unstable (currently 105.0.5191.2-1). No change.

I switched to another machine, also running a fully updated Debian Bullseye. It just worked. I cleared out the cache/profile directories to start fresh and followed the above instructions. And it just worked. I really have no idea what could be different here.

**Comment by neilschelly** [comment]
I was able to make it work on the first machine. I undocked it, so it was _just_ a laptop without my external speakers, keyboard, mouse, and extra monitor.

I opened the brand new Chrome browser window and logged into my profile this time. I got to the error above by just trying to setup FlowCrypt as normal. I unplugged my dock, and I re-used that same window. I tried again, and it brought up the oAuth window without complaint. I switched back to my dock after authenticating to Google, but before entering my private key. It wouldn't let me continue; same error as above.

I needed to finish the setup, not just the oAuth login piece while undocked. Once I was finished, FlowCrypt works as expected regardless of whether or not I'm docked.

Hope that helps!
-Neil

**Comment by martgil** [comment]
Hello @neilschelly Thanks for the additional information and sorry if we take the time to respond on this one. When I read about the visible screen size error, I tried interacting with the browser extension with the smaller screen and none of them work back then. And the recent info about the docked setup with external/extra monitor, I started to think that it might be about the external monitor?

I found an article that might be related to what you've been experiencing - https://superuser.com/questions/1029342/chrome-extension-popups-not-displaying-on-second-monitor. At this point, you might already figured out the solution, but if not, let's try unchecking the hardware acceleration option on the Google Chrome settings (chrome:settings -> search for hardware acceleration -> disable it -> try logging into your docked setup). 

Some other supporting materials: https://www.reddit.com/r/chrome/comments/3om8pq/extensions_dont_respond_on_second_monitor_happens/

Thank you for your patience,
Mart

**Comment by neilschelly** [comment]
I just tried disabling the hardware acceleration, but to no avail. I got exactly the same error as originally reported above.

**Comment by martgil** [comment]
Hello, @neilschelly thanks for trying that out. By the way, I am finally able to produce the issue. In my testing case, the size of the OAuth window seemed to be too big for the display output.. something like that.

https://user-images.githubusercontent.com/46025304/181166620-7e5974e9-37fd-4efe-ab1c-2ea726ba8cad.mov

I made some changes from my local source code and built a debug version for you to try on. I'll be sending them over email. Thank you!

**Comment by neilschelly** [comment]
That is exactly the behavior I've been seeing. For reference, my docked setup includes two monitors. The left is my laptop screen (1920x1080) and the right is my external monitor (3840x2160). They are aligned at the bottom, so there is a "gap" in the overall rectangular area above the laptop screen.

**Comment by neilschelly** [comment]
I ran the debug version just now. Same result. I do see errors next to the extension now that I can navigate visually, but I cannot see an easy way to export the logs to share. I'm not familiar with Chrome extension debugging, but I'm looking for something like that.

**Comment by neilschelly** [comment]
```catch.js:60 Error: failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
    at Catch.nameAndDetailsAsException (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/js/common/platform/catch.js:345:15)
    at Catch.report (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/js/common/platform/catch.js:98:34)
    at Settings.newGoogleAcctAuthPromptThenAlertOrForward (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/js/common/settings.js:361:23)
    at async SettingsView.<anonymous> (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/chrome/settings/index.js:201:28)

details:
[typeof:object:[object Object]] {"result":"Error","error":"No oauth window returned after initiating it","acctEmail":"neil@jenandneil.com"}
Catch.onErrorInternalHandler @ catch.js:60
Catch.reportErr @ catch.js:92
Catch.report @ catch.js:98
Settings.newGoogleAcctAuthPromptThenAlertOrForward @ settings.js:361
await in Settings.newGoogleAcctAuthPromptThenAlertOrForward (async)
(anonymous) @ index.js:201
cbWithErrsHandled @ ui.js:74
(anonymous) @ ui.js:110
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
catch.js:64 failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
```

```Error: failed to log into google in newGoogleAcctAuthPromptThenAlertOrForward
    at Catch.nameAndDetailsAsException (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/js/common/platform/catch.js:345:15)
    at Catch.report (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/js/common/platform/catch.js:98:34)
    at Settings.newGoogleAcctAuthPromptThenAlertOrForward (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/js/common/settings.js:361:23)
    at async SettingsView.<anonymous> (chrome-extension://mefhmkjnejkbdefeflagngpmcdpiadkf/chrome/settings/index.js:201:28)

details:
[typeof:object:[object Object]] {"result":"Error","error":"No oauth window returned after initiating it","acctEmail":"neil@jenandneil.com"}
Catch.onErrorInternalHandler @ catch.js:64
Catch.reportErr @ catch.js:92
Catch.report @ catch.js:98
Settings.newGoogleAcctAuthPromptThenAlertOrForward @ settings.js:361
await in Settings.newGoogleAcctAuthPromptThenAlertOrForward (async)
(anonymous) @ index.js:201
cbWithErrsHandled @ ui.js:74
(anonymous) @ ui.js:110
dispatch @ jquery.min.js:2
v.handle @ jquery.min.js:2
```

```index.htm:1 Unchecked runtime.lastError: Invalid value for bounds. Bounds must be at least 50% within visible screen space.```

**Comment by martgil** [comment]
@neilschelly Thank you for sharing even more helpful info.

@rrrooommmaaa Greetings, sir! Do you have any idea about this issue? 

Based on my testing case, it is possibly related to spawning the OAuth window to an out-of-bounds position. I tried changing the size of the OAuth window to 90% of the screen size but it doesn't work. I had a new idea of changing the spawning points of the OAuth to the center of the active display since Neil is using a dual monitor setup.

**Comment by neilschelly** [comment]
I just installed the new debug version you sent me, and it worked fine. The oAuth window popped up immediately, as expected.
-Neil

**Comment by rrrooommmaaa** [comment]
> index.htm:1 Unchecked runtime.lastError: Invalid value for bounds. Bounds must be at least 50% within visible screen space.

> They are aligned at the bottom, so there is a "gap" in the overall rectangular area above the laptop screen.

The idea that the gap is the culprit seems interesting to me.
If it's true, then  `const oauthWin = await windowsCreate({ url, left: 100, top: 50, height: 800, width: 550, type: 'popup' })` 
is possibly trying to position the Oauth popup somewhere here:
![image](https://user-images.githubusercontent.com/23335944/181572377-af75369a-c6f7-4bfd-a415-7c510669fe13.png)
and its 50% is outside of the visible space.

And, I guess, this setup doesn't fail?
![image](https://user-images.githubusercontent.com/23335944/181572859-391021f2-9a1c-438c-83c7-6f87851b6f6a.png)
If so, the correct solution would be to fix the code to have `left: baseX + 100, top: baseY + 50`
where `baseX` and `baseY` reflect the actual boundaries...

@martgil Can you create a draft PR with your updates so I can have a look?

**Comment by neilschelly** [comment]
I just confirmed @rrrooommmaaa 's theory. I switched my screen layout to the second one he suggested, reinstalled the current release version of the extension, and it worked. I'm "working" on the right screen in that diagram, but the oAuth window is popping up on the left one now. And based on those coordinates above, I can only assume that window would have been popping up at coordinates that don't exist in my setup.
